packages/components/button/src/button.vue

```diff
@@ -25,31 +25,23 @@
 
 <script lang="ts">
 import { computed, inject, defineComponent } from 'vue'
-import { elFormKey, elFormItemKey } from '@element-plus/tokens'
-import { useGlobalConfig } from '@element-plus/utils/util'
+import { useFormItem } from '@element-plus/hooks'
 import { elButtonGroupKey } from '@element-plus/tokens'
 
+import type { ComponentSize } from '@element-plus/utils/types'
 import { buttonEmits, buttonProps } from './button'
 
-import type { ElFormContext, ElFormItemContext } from '@element-plus/tokens'
-
 export default defineComponent({
   name: 'ElButton',
 
   props: buttonProps,
   emits: buttonEmits,
 
-  setup(props, { emit }) {
-    const $ELEMENT = useGlobalConfig()
-
-    const elForm = inject<ElFormContext>(elFormKey)
-    const elFormItem = inject<ElFormItemContext>(elFormItemKey)
-    const elBtnGroup = inject(elButtonGroupKey)
-
-    const buttonSize = computed(
-      () => props.size || elBtnGroup?.size || elFormItem?.size || $ELEMENT.size
-    )
-    const buttonDisabled = computed(() => props.disabled || elForm?.disabled)
+  setup(_, { emit }) {
+    const elBtnGroup = inject(elButtonGroupKey, undefined)
+    const { size: buttonSize, disabled: buttonDisabled } = useFormItem({
+      size: computed(() => elBtnGroup?.size as ComponentSize),
```

Comment:
```suggestion
      size: computed(() => elBtnGroup?.size),
```
I think `as ComponentSize` can be removed.

Reply:
No.  since the size here is type `undefined | ComponentSize` if you remove the cast the assertion would fail because undefined is not type of `ComponentSize`

Reply:
Emmm, you're right. But I think you should change `LocalFallbacks['size']` as `Ref<IUseFormItemProps['size'] | undefined>`.

Reply:
I think the question mark is already marking the member as optional so that we don't need to add `undefined` annotation which seems to be redundant.

Reply:
In fact, `elBtnGroup?.size` could be `undefined`.

Reply:
The use of forced type conversion here is not in compliance with the specification.

Reply:
Yes, and that is acceptable since we have handled the situation in `use-form-item`. But it seems like the compiler is acting weird which I think the syntax here is correct by marking the member optional should provide the type `undefined` for it.

Reply:
Never mind. I will handle this tomorrow...